core/range/reverse_each_spec.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "3.3" do
+  describe "Range#reverse_each" do
+    it "works efficiently for very long Ranges of Integers" do
+      (1..2**100).reverse_each.take(3).size.should == 3
+    end
+
+    it "works for infinite Ranges of Integers" do
+      (..5).reverse_each.take(3).should == [5, 4, 3]
+    end
+
+    it "works for Ranges of Strings by converting the Range to an Array first" do
+      ("a".."z").reverse_each.take(3).should == ["z", "y", "x"]
+    end
+
+    it "raises a TypeError for endless Ranges of Integers" do
+      -> {
+        (1..).reverse_each.take(3)
+      }.should raise_error(TypeError, "can't iterate from NilClass")
+    end
+
+    it "raises a TypeError for endless Ranges of other objects" do
+      -> {
+        ("a"..).reverse_each.take(3)
+      }.should raise_error(TypeError, "can't iterate from NilClass")
+    end
+  end
+end
```

Comment:
It makes sense to check the following files and have here the similar cases/structure:
- `core/range/each_spec.rb`
- `core/enumerable/reverse_each_spec.rb`
- `core/array/reverse_each_spec.rb`

Reply:
All these tests are pretty specific for ranges. There is a spec in `core/range/each_spec` that validates the behaviour for a beginless range:
```ruby
it "raises a TypeError beginless ranges" do
  -> { (..2).each { |x| x } }.should raise_error(TypeError)
end
```
This specific case is not relevant for arrays, you can simply reverse iterate over it, there is no need there to construct a temporary array and there is no chance for it to become infinite.

As for `Enumerable#reverse_each`: that depends on the object that's being enumerated. If the input is an infinite generator, it will probably result in memory exhaustion